diff --git a/src/LayoutsManager.vala b/src/LayoutsManager.vala
index 1bac80e..7301fbf 100644
--- a/src/LayoutsManager.vala
+++ b/src/LayoutsManager.vala
@@ -97,7 +97,7 @@ public class Keyboard.Widgets.LayoutManager : Gtk.ScrolledWindow {
 
     public string? get_name_for_xkb_layout (string language, string? variant) {
         debug ("get_name_for_xkb_layout (%s, %s)", language, variant);
-        Xml.Doc* doc = Xml.Parser.parse_file ("/usr/share/X11/xkb/rules/evdev.xml");
+        Xml.Doc* doc = Xml.Parser.parse_file ("@xlibs@");
         if (doc == null) {
             critical ("'evdev.xml' not found or permissions incorrect\n");
             return null;
