diff --git a/src/Agent.vala b/src/Agent.vala
index cee29e0..0627b17 100644
--- a/src/Agent.vala
+++ b/src/Agent.vala
@@ -34,7 +34,9 @@ namespace Ag {
 
         construct {
             application_id = "io.elementary.desktop.agent-geoclue2";
-            settings = new Settings (application_id);
+            SettingsSchemaSource sss = new SettingsSchemaSource.from_directory ("@PANTHEON_AGENT_GEOCLUE2_GSETTINGS_PATH@", SettingsSchemaSource.get_default (), true);
+            SettingsSchema geoclue_schema = sss.lookup (application_id, false);
+            settings = new Settings.full (geoclue_schema, null, null);
 
             settings.changed.connect ((key) => {
                 refresh_enabled_state ();
