diff --git a/src/Plug.vala b/src/Plug.vala
index ff85fd0..1d9be37 100644
--- a/src/Plug.vala
+++ b/src/Plug.vala
@@ -25,7 +25,6 @@ namespace Power {
     public class Plug : Switchboard.Plug {
         private Gtk.SizeGroup label_size;
         private Gtk.StackSwitcher stack_switcher;
-        private GLib.Settings elementary_dpms_settings;
 
         private PowerSettings screen;
         private Battery battery;
@@ -51,8 +50,7 @@ namespace Power {
         public override Gtk.Widget get_widget () {
             if (stack_container == null) {
                 settings = new GLib.Settings ("org.gnome.settings-daemon.plugins.power");
-                elementary_dpms_settings = new GLib.Settings ("io.elementary.dpms");
-    
+
                 battery = new Battery ();
                 power_supply = new PowerSupply ();
     
@@ -297,25 +295,15 @@ namespace Power {
                 main_grid.attach (lock_image, 2, 6, 1, 1);
             }
 
-            var screen_timeout_label = new Gtk.Label (_("Turn off display when inactive for:"));
-            screen_timeout_label.halign = Gtk.Align.END;
-            screen_timeout_label.xalign = 1;
-
-            var screen_timeout = new TimeoutComboBox (elementary_dpms_settings, "standby-time");
-            screen_timeout.changed.connect (run_dpms_helper);
-
             var power_label = new Gtk.Label (_("Power button:"));
             power_label.halign = Gtk.Align.END;
             power_label.xalign = 1;
 
             var power_combobox = new ActionComboBox ("power-button-action");
 
-            main_grid.attach (screen_timeout_label, 0, 3, 1, 1);
-            main_grid.attach (screen_timeout, 1, 3, 1, 1);
-            main_grid.attach (power_label, 0, 4, 1, 1);
-            main_grid.attach (power_combobox, 1, 4, 1, 1);
+            main_grid.attach (power_label, 0, 3, 1, 1);
+            main_grid.attach (power_combobox, 1, 3, 1, 1);
 
-            label_size.add_widget (screen_timeout_label);
             label_size.add_widget (power_label);
 
             return main_grid;
@@ -418,17 +406,6 @@ namespace Power {
 
             return false;
         }
-
-        private static void run_dpms_helper () {
-            try {
-                string[] argv = { "elementary-dpms-helper" };
-                Process.spawn_async (null, argv, Environ.get (),
-                    SpawnFlags.SEARCH_PATH | SpawnFlags.STDERR_TO_DEV_NULL | SpawnFlags.STDOUT_TO_DEV_NULL,
-                    null, null);
-            } catch (SpawnError e) {
-                warning ("Failed to reset dpms settings: %s", e.message);
-            }
-        }
     }
 }
 
