diff --git a/CMakeLists.txt b/CMakeLists.txt
index f64eabd..886cd12 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,7 +12,7 @@ set (GSETTINGSDIR "${DATADIR}/glib-2.0/schemas/")
 set (PKGDATADIR "${DATADIR}/${CMAKE_PROJECT_NAME}")
 set (GETTEXT_PACKAGE "${CMAKE_PROJECT_NAME}")
 set (RELEASE_NAME "Wingpanel")
-set (INDICATORS_DIR ${CMAKE_INSTALL_FULL_LIBDIR}/${CMAKE_PROJECT_NAME})
+set (INDICATORS_DIR "/run/current-system/sw/lib/wingpanel")
 set (VERSION "2.1.1")
 set (VERSION_INFO "Stable Release")
 
diff --git a/lib/wingpanel.pc.cmake b/lib/wingpanel.pc.cmake
index 1c19b30..6981466 100644
--- a/lib/wingpanel.pc.cmake
+++ b/lib/wingpanel.pc.cmake
@@ -1,8 +1,8 @@
 prefix=@PREFIX@
 exec_prefix=${prefix}
-libdir=${prefix}/@CMAKE_INSTALL_LIBDIR@
-includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@
-indicatorsdir=${libdir}/wingpanel
+libdir=@CMAKE_INSTALL_LIBDIR@
+includedir=@CMAKE_INSTALL_INCLUDEDIR@
+indicatorsdir=/run/current-system/sw/lib/wingpanel
  
 Name: Wingpanel
 Description: Wingpanel headers
