diff --git a/autostart/gnome-keyring-gpg-pantheon.desktop b/autostart/gnome-keyring-gpg-pantheon.desktop
deleted file mode 100644
index 059704d..0000000
--- a/autostart/gnome-keyring-gpg-pantheon.desktop
+++ /dev/null
@@ -1,15 +0,0 @@
-[Desktop Entry]
-Type=Application
-Name=GPG Password Agent
-Comment=GNOME Keyring: GPG Agent
-Exec=/usr/bin/gnome-keyring-daemon --start --components=gpg
-OnlyShowIn=Pantheon;
-X-GNOME-Autostart-Phase=Initialization
-X-GNOME-AutoRestart=false
-X-GNOME-Autostart-Notify=true
-X-GNOME-Bugzilla-Bugzilla=GNOME
-X-GNOME-Bugzilla-Product=gnome-keyring
-X-GNOME-Bugzilla-Component=general
-X-GNOME-Bugzilla-Version=3.10.1
-NoDisplay=true
-X-Ubuntu-Gettext-Domain=gnome-keyring
diff --git a/autostart/gnome-keyring-pkcs11-pantheon.desktop b/autostart/gnome-keyring-pkcs11-pantheon.desktop
index 824f855..bed4932 100644
--- a/autostart/gnome-keyring-pkcs11-pantheon.desktop
+++ b/autostart/gnome-keyring-pkcs11-pantheon.desktop
@@ -1,15 +1,153 @@
 [Desktop Entry]
 Type=Application
+Name[af]=Sertifikaat- en sleutelberging
+Name[ar]=مخزن الشهادات والمفاتيح
+Name[as]=প্ৰমাণপত্ৰ আৰু কি সংৰক্ষণ
+Name[ast]=Certificaos y almacenamientu de claves
+Name[be]=Сховішча сертыфікатаў і ключоў
+Name[bg]=Съхранение на сертификати и ключове
+Name[bn]=সার্টিফিকেট এবং কী স্টোরেজ
+Name[bn_IN]=সার্টিফিকেট এবং কি সংগ্রহস্থল
+Name[bs]=Skladište sertifikata i ključeva
+Name[ca]=Magatzem de certificats i de claus
+Name[ca@valencia]=Magatzem de certificats i de claus
+Name[cs]=Odemknout certifikáty a umístění klíčů
+Name[da]=Certifikat- og nøglelager
+Name[de]=Zertifikat-/Schlüsselverwahrung
+Name[el]=Αποθήκη πιστοποιητικών και κλειδιών
+Name[en_GB]=Certificate and Key Storage
+Name[eo]=Memorejo de atestiloj kaj ŝlosiloj
+Name[es]=Certificados y almacenamiento de claves
+Name[et]=Sertifikaadi- ja võtmehoidla
+Name[eu]=Ziurtagirien/gakoen biltegia
+Name[fa]=مخزن گواهی‌نامه و کلید
+Name[fi]=Varmenne- ja avainsäilö
+Name[fr]=Stockage de certificats et de clés
+Name[fur]=Archivi certificâts e clâfs
+Name[gd]=Teisteanas agus stòras na h-iuchrach
+Name[gl]=Almacenamento de chave ou certificado
+Name[gu]=પ્રમાણપત્ર અને કી સંગ્રહ
+Name[he]=אחסון תעודה ומפתח
+Name[hi]=कुंजी भंडार प्रमाणित करें
+Name[hr]=Vjerodajnice i spremište ključeva
+Name[hu]=Tanúsítvány- és kulcstároló
+Name[id]=Penyimpanan Kunci dan Sertifikat
+Name[it]=Archivio certificati e chiavi
+Name[ja]=証明書および鍵を格納するストレージ
+Name[kk]=Сертификаттар және кілттер қоймасы
+Name[km]=វិញ្ញាបនបត្រ និងការ​ផ្ទុកសោ
+Name[kn]=ಪ್ರಮಾಣಪತ್ರ ಮತ್ತು ಕೀಲಿಯ ಶೇಖರಣೆ
+Name[ko]=인증서 및 키 저장소
+Name[lt]=Liudijimų ir raktų saugykla
+Name[lv]=Sertifikātu un atslēgu glabātava
+Name[mk]=Склад на клучеви и сертификати
+Name[ml]=സാക്ഷ്യപത്രവും കീ സംഭരണവും
+Name[mr]=प्रमाणपत्र व कि स्टोरेज
+Name[ms]=Sijil dan Penyimpanan Kekunci
+Name[nb]=Lager for sertifikat/nøkkel
+Name[nl]=Certificaat- en sleutelopslag
+Name[nn]=Serifikat- og nøkkellager
+Name[oc]=Emmagazinatge de certificats e de claus
+Name[or]=ପ୍ରମାଣପତ୍ର ଏବଂ କି ସଂରକ୍ଷଣ
+Name[pa]=ਸਰਟੀਫਿਕੇਟ ਅਤੇ ਕੁੰਜੀ ਸਟੋਰੇਜ਼
+Name[pl]=Przechowalnia certyfikatów i kluczy
+Name[pt]=Armazenamento de chaves e certificados
+Name[pt_BR]=Certificados e armazenamento de chaves
+Name[ro]=Stocare pentru certificate și chei
+Name[ru]=Хранилище сертификатов и ключей
+Name[sk]=Úložisko certifikátov a kľúčov
+Name[sl]=Hranjenje potrdil in ključev
+Name[sr]=Складиште сертификата и кључева
+Name[sr@latin]=Skladište sertifikata i ključeva
+Name[sv]=Certifikat och nyckellagring
+Name[ta]=சான்றிதழ்/விசை சேமிப்பகம்
+Name[te]=ధృవీకరణపత్రం మరియు కీ నిల్వ
+Name[th]=แหล่งใบรับรองและกุญแจ
+Name[tr]=Sertifika ve Anahtar Deposu
+Name[ug]=گۇۋاھنامە ۋە شىفىرلىق ئاچقۇچ ئامبىرى
+Name[uk]=Сертифікат та сховище ключів
+Name[vi]=Lưu trữ Chứng nhận và khóa
+Name[zh_CN]=证书和密钥存储
+Name[zh_HK]=證書與密碼匙儲存區
+Name[zh_TW]=憑證與金鑰儲存區
 Name=Certificate and Key Storage
+Comment[af]=GNOME-sleutelring: PKCS#11-komponent
+Comment[ar]=حلقة مفاتيح جنوم: مكون PKCS#11
+Comment[as]=GNOME Keyring: PKCS#11 উপাদান
+Comment[ast]=Depósitu de claves de GNOME: Componente PKCS#11
+Comment[be]=Вязкі ключоў GNOME: кампанент PKCS#11
+Comment[bg]=Ключодържател на GNOME: компонент за PKCS#11
+Comment[bn]=GNOME কী রিং: PKCS#11 কম্পোনেন্ট
+Comment[bn_IN]=GNOME Keyring: PKCS#11 উপাদান
+Comment[bs]=Gnomovi privjesci: PKCS#11 komponenta
+Comment[ca]=Anell de claus del GNOME: component PKCS#11
+Comment[ca@valencia]=Anell de claus del GNOME: component PKCS#11
+Comment[cs]=Klíčenka GNOME: komponenta PKCS#11
+Comment[da]=GNOME-nøgleringsdæmon: PKCS#11-komponent
+Comment[de]=GNOME-Schlüsselbunddienst: PKCS#11-Komponente
+Comment[el]=Κλειδοθήκη GNOME: Συστατικό στοιχείο PKCS#11
+Comment[en_GB]=GNOME Keyring: PKCS#11 Component
+Comment[es]=Depósito de claves de GNOME: Componente PKCS#11
+Comment[et]=GNOME võtmerõngas: PKCS#11 komponent
+Comment[eu]=GNOMEren gako-sorta: PKCS#11 osagaia
+Comment[fa]=دسته‌کلید گنوم: عناصر PKCS#11
+Comment[fi]=Gnomen avainnippu: PKCS#11-komponentti
+Comment[fr]=Trousseau de clés de GNOME : composant PKCS#11
+Comment[fur]=Puarteclâfs di GNOME: component PKCS#11
+Comment[gd]=Dul-iuchrach GNOME: Co-phàirt PKCS#11
+Comment[gl]=GNOME Keyring: compoñente PKCS#11
+Comment[gu]=GNOME કીરીંગ: PKCS#11 ઘટક
+Comment[he]=קבוצת מפתחות של GNOME:רכיב PKCS#11
+Comment[hi]=गनोम कीरिंग: PKCS#11 घटक
+Comment[hr]=GNOME skup ključeva: PKCS#11 komponena
+Comment[hu]=GNOME kulcstartó – PKCS#11 összetevő
+Comment[id]=GNOME Keyring: Komponen PKCS#11
+Comment[it]=Portachiavi di GNOME: componente PKCS#11
+Comment[ja]=GNOME キーリング: PKCS#11 コンポーネント
+Comment[kk]=GNOME Keyring: PKCS#11 құрамасы
+Comment[km]=GNOME Keyring ៖ សមាសភាគ PKCS#11
+Comment[kn]=GNOME ಕೀಲಿಗೊಂಚಲು: PKCS#11 ಘಟಕ
+Comment[ko]=그놈 키 모음: PKCS#11 컴포넌트
+Comment[lt]=GNOME raktinė: PKCS#11 komponentas
+Comment[lv]=GNOME atslēgu saišķis — PKCS#11 komponente
+Comment[mk]=Приврзок на GNOME: PKCS#11 компонента
+Comment[ml]=ഗ്നോം കീറിങ്: PKCS#11 ഘടകം
+Comment[mr]=GNOME किरिंग: PKCS#11 घटक
+Comment[ms]=GNOME Keyring: Komponen PKCS#11
+Comment[nb]=GNOME nøkkelring: PKCS#11-komponent
+Comment[nl]=Sleutelbos-service: PKCS#11-component
+Comment[nn]=GNOME Nøkkelring: PKCS#11-komponent
+Comment[oc]=Trossèl de claus GNOME : component PKCS#11
+Comment[or]=GNOME କି ରିଙ୍ଗ: PKCS#11 ଉପାଦାନ
+Comment[pa]=ਗਨੋਮ ਕੀਰਿੰਗ: PKCS#11 ਭਾਗ
+Comment[pl]=Baza kluczy dla środowiska GNOME: składnik PKCS#11
+Comment[pt]=GNOME Keyring: componente PKCS#11
+Comment[pt_BR]=Chaveiro do GNOME: Componente PKCS#11
+Comment[ro]=Inelul de chei GNOME: Componenta PKCS#11
+Comment[ru]=Связка ключей GNOME: компонент PKCS#11
+Comment[sk]=Modul PKCS#11 zväzku kľúčov GNOME
+Comment[sl]=Zbirka ključev GNOME: enota PKCS#11
+Comment[sr]=Гномови привесци: ПКЦС#11 компонента
+Comment[sr@latin]=Gnomovi privesci: PKCS#11 komponenta
+Comment[sv]=GNOME-nyckelring: PKCS#11-komponent
+Comment[ta]=GNOME கீரிங்:  PKCS#11 கூறு
+Comment[te]=GNOME కీరింగ్: PKCS#11 అంశం
+Comment[th]=พวงกุญแจของ GNOME: องค์ประกอบ PKCS#11
+Comment[tr]=GNOME Anahtarlığı: PKCS#11 Bileşeni
+Comment[ug]=گىنوم ئاچقۇچ ھالقىسى：PKCS#11 دېتالى
+Comment[uk]=Служба в'язки ключів GNOME: компонент PKCS#11
+Comment[vi]=Chùm chìa khóa GNOME: thành phần PKCS#11
+Comment[zh_CN]=GNOME 密钥环：PKCS#11 组件
+Comment[zh_HK]=GNOME 密碼匙圈：PKCS#11 元件
+Comment[zh_TW]=GNOME 鑰匙圈：PKCS#11 元件
 Comment=GNOME Keyring: PKCS#11 Component
-Exec=/usr/bin/gnome-keyring-daemon --start --components=pkcs11
-OnlyShowIn=Pantheon;
-X-GNOME-Autostart-Phase=Initialization
+Exec=@keyring@/gnome-keyring-daemon --start --components=pkcs11
+OnlyShowIn=GNOME;Unity;MATE;Pantheon;
+NoDisplay=true
+X-GNOME-Autostart-Phase=PreDisplayServer
 X-GNOME-AutoRestart=false
 X-GNOME-Autostart-Notify=true
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-keyring
 X-GNOME-Bugzilla-Component=general
-X-GNOME-Bugzilla-Version=3.10.1
-NoDisplay=true
-X-Ubuntu-Gettext-Domain=gnome-keyring
+X-GNOME-Bugzilla-Version=3.28.2
diff --git a/autostart/gnome-keyring-secrets-pantheon.desktop b/autostart/gnome-keyring-secrets-pantheon.desktop
deleted file mode 100644
index 63590ba..0000000
--- a/autostart/gnome-keyring-secrets-pantheon.desktop
+++ /dev/null
@@ -1,15 +0,0 @@
-[Desktop Entry]
-Type=Application
-Name=Secret Storage Service
-Comment=GNOME Keyring: Secret Service
-Exec=/usr/bin/gnome-keyring-daemon --start --components=secrets
-OnlyShowIn=Pantheon;
-X-GNOME-Autostart-Phase=Initialization
-X-GNOME-AutoRestart=false
-X-GNOME-Autostart-Notify=true
-X-GNOME-Bugzilla-Bugzilla=GNOME
-X-GNOME-Bugzilla-Product=gnome-keyring
-X-GNOME-Bugzilla-Component=general
-X-GNOME-Bugzilla-Version=3.10.1
-NoDisplay=true
-X-Ubuntu-Gettext-Domain=gnome-keyring
diff --git a/autostart/gnome-keyring-secrets-patheon.desktop b/autostart/gnome-keyring-secrets-patheon.desktop
new file mode 100644
index 0000000..e4fa735
--- /dev/null
+++ b/autostart/gnome-keyring-secrets-patheon.desktop
@@ -0,0 +1,154 @@
+[Desktop Entry]
+Type=Application
+Name[af]=Geheimbergingsdiens
+Name[ar]=خدمة حفظ سرية
+Name[as]=গুপ্ত সংৰক্ষণ সেৱা
+Name[ast]=Serviciu d'almacenamientu de secretos
+Name[be]=Служба сховішча для сакрэтаў
+Name[bg]=Услуга за шифриран носител
+Name[bn]=গোপন সংরক্ষণ সম্পর্কিত সার্ভিস
+Name[bn_IN]=গোপন সংগ্রহস্থল পরিষেবা
+Name[bs]=Servis sa tajno skladištenje
+Name[ca]=Servei d'emmagatzematge de secrets
+Name[ca@valencia]=Servei d'emmagatzematge de secrets
+Name[cs]=Služba utajení dat
+Name[da]=Secret Storage Service
+Name[de]=Sicherheitsdienst
+Name[el]=Υπηρεσία κρυφής αποθήκευσης
+Name[en_GB]=Secret Storage Service
+Name[eo]=Servo pri sekreta memorejo
+Name[es]=Servicio de almacenamiento de secretos
+Name[et]=Saladuste hoidmise teenus
+Name[eu]=Ezkutuko biltegiaren zerbitzua
+Name[fa]=سرویس مخزن محرمانه
+Name[fi]=Salaisuuksien säilöntäpalvelu
+Name[fr]=Service de stockage secret
+Name[fur]=Servizi archivi segret
+Name[gd]=Seirbheis stòrais dhìomhair
+Name[gl]=Servizo de almacenamento segredo
+Name[gu]=ખાનગી સંગ્રહ સેવા
+Name[he]=שרות אחסון חשאי
+Name[hi]=गुप्त भंडार सेवा
+Name[hr]=Tajna usluga spremišta
+Name[hu]=Titoktároló szolgáltatás
+Name[id]=Layanan Penyimpanan Rahasia
+Name[it]=Servizio archivio segreto
+Name[ja]=シークレットストレージサービス
+Name[kk]=Құпия кілттер қоймасы қызметі
+Name[km]=សេវា​ផ្ទុក​សម្ងាត់
+Name[kn]=ಸೀಕ್ರೆಟ್ ಶೇಖರಣಾ ಸೇವೆ
+Name[ko]=비밀 저장고 서비스
+Name[lt]=Slaptoji saugojimo tarnyba
+Name[lv]=Slepenās glabātavas serviss
+Name[mk]=Сервис за склад на тајни
+Name[ml]=രഹസ്യ സംഭരണ സേവനം
+Name[mr]=गोपणी स्टोरेज सर्व्हिस
+Name[ms]=Servis Rahsia Penyimpanan
+Name[nb]=Lagringstjeneste for hemmeligheter
+Name[nl]=Secret Storage Service
+Name[nn]=Hemmeleg lagerteneste
+Name[oc]=Servici d'emmagazinatge secret
+Name[or]=ଗୁପ୍ତ ସଂରକ୍ଷଣ ସର୍ଭିସ
+Name[pa]=ਗੁਪਤ ਸਟੋਰੇਜ਼ ਸਰਵਿਸ
+Name[pl]=Usługa przechowywania haseł
+Name[pt]=Serviço de armazenamento secreto
+Name[pt_BR]=Serviço de armazenamento secreto
+Name[ro]=Serviciul secret de stocare
+Name[ru]=Служба безопасного хранения
+Name[sk]=Služba utajeného úložiska
+Name[sl]=Shranjevanje skritih storitev
+Name[sr]=Сервис са тајно складиштење
+Name[sr@latin]=Servis sa tajno skladištenje
+Name[sv]=Lagringstjänst för hemligheter
+Name[ta]=\sரகசிய சேமிப்பு சேவை
+Name[te]=రహస్య నిల్వ సేవ
+Name[th]=บริการเก็บข้อมูลลับ
+Name[tr]=Gizli Depolama Hizmeti
+Name[ug]=مەخپىي ساقلاش مۇلازىمىتى
+Name[uk]=Служба шифрування дисків
+Name[vi]=Dịch vụ lưu trữ bảo mật
+Name[zh_CN]=保密存储服务
+Name[zh_HK]=保密儲存區服務
+Name[zh_TW]=保密儲存區服務
+Name=Secret Storage Service
+Comment[af]=GNOME-sleutelring: geheimdiens
+Comment[ar]=حلقة مفاتيح جنوم: خدمة سرية
+Comment[as]=GNOME Keyring: গুপ্ত সেৱা
+Comment[ast]=Depósitu de claves de GNOME: Serviciu de secretos
+Comment[be]=Вязкі ключоў GNOME: служба сакрэтаў
+Comment[bg]=Ключодържател на GNOME: услуга за тайни
+Comment[bn]=GNOME কী রিং: গোপন সার্ভিস
+Comment[bn_IN]=GNOME Keyring: গোপন পরিষেবা
+Comment[bs]=Gnomovi privjesci: servis za tajnost
+Comment[ca]=Anell de claus del GNOME: servei de secrets
+Comment[ca@valencia]=Anell de claus del GNOME: servei de secrets
+Comment[cs]=Klíčenka GNOME: Služba utajení dat
+Comment[da]=GNOME-nøglering: Secret Service
+Comment[de]=GNOME-Schlüsselbunddienst: Sicherheitsdienst
+Comment[el]=Κλειδοθήκη GNOME: Κρυφή υπηρεσία
+Comment[en_GB]=GNOME Keyring: Secret Service
+Comment[es]=Depósito de claves de GNOME: Servicio de secretos
+Comment[et]=GNOME võtmerõngas: saladuste teenus
+Comment[eu]=GNOMEren gako-sorta: ezkutuko zerbitzua
+Comment[fa]=دسته‌کلید گنوم: سرویس محرمانه
+Comment[fi]=Gnomen avainnippu: Salainen palvelu
+Comment[fr]=Trousseau de clés de GNOME : service secret
+Comment[fur]=Puarteclâfs di GNOME: servizi segret
+Comment[gd]=Dul-iuchrach GNOME: Seirbheis dhìomhair
+Comment[gl]=GNOME Keyring: servizo segredo
+Comment[gu]=GNOME કીરીંગ: ખાનગી સેવા
+Comment[he]=קבוצת מפתחות של GNOME: שירות חשאי
+Comment[hi]=गनोम कीरिंग: गुप्त सेवा
+Comment[hr]=GNOME skup ključeva: Tajna usluga
+Comment[hu]=GNOME kulcstartó – titokszolgáltatás
+Comment[id]=GNOME Keyring: Layanan Rahasia
+Comment[it]=Portachiavi di GNOME: servizio segreto
+Comment[ja]=GNOME キーリング: シークレットサービス
+Comment[kk]=GNOME Keyring: Құпия кілттер қызметі
+Comment[km]=GNOME Keyring ៖ សេវា​សម្ងាត់
+Comment[kn]=GNOME ಕೀಲಿಗೊಂಚಲು: ಸೀಕ್ರೆಟ್ ಸೇವೆ
+Comment[ko]=그놈 키 모음: 비밀 서비스
+Comment[lt]=GNOME raktinė: slaptoji tarnyba
+Comment[lv]=GNOME atslēgu saišķis — slepenais serviss
+Comment[mk]=Приврзок на GNOME: таен сервис
+Comment[ml]=ഗ്നോം കീറിങ്: രഹസ്യ സേവനം
+Comment[mr]=GNOME किरिंग: गोपणीय सर्व्हिस
+Comment[ms]=GNOME Keyring: Servis Rahsia
+Comment[nb]=GNOME nøkkelring: Hemmelige tjenester
+Comment[ne]=जिनोम किरिङ : गुप्त सेवा
+Comment[nl]=Sleutelbos-service: Secret Service
+Comment[nn]=GNOME Nøkkelring: Hemmeleg teneste
+Comment[oc]=Trossèl de claus GNOME : servici secret
+Comment[or]=GNOME କି ରିଙ୍ଗ: ଗୁପ୍ତ ସଂରକ୍ଷଣ
+Comment[pa]=ਗਨੋਮ ਕੀਰਿੰਗ ਡੈਮਨ: ਗੁਪਤ ਸਰਵਿਸ
+Comment[pl]=Baza kluczy dla środowiska GNOME: usługa haseł
+Comment[pt]=GNOME Keyring: serviço secreto
+Comment[pt_BR]=Chaveiro do GNOME: Serviço secreto
+Comment[ro]=Inelul de chei GNOME: Serviciul secret
+Comment[ru]=Связка ключей GNOME: служба безопасности
+Comment[sk]=Služba utajenia zväzku kľúčov GNOME
+Comment[sl]=Zbirka ključev GNOME: skrite storitve
+Comment[sr]=Гномови привесци: сервис за тајност
+Comment[sr@latin]=Gnomovi privesci: servis za tajnost
+Comment[sv]=GNOME-nyckelring: Hemlighetstjänst
+Comment[ta]=GNOME கீரிங்: ரகசிய சேவை
+Comment[te]=GNOME కీరింగ్: రహస్య సేవ
+Comment[th]=พวงกุญแจของ GNOME: บริการเก็บข้อมูลลับ
+Comment[tr]=GNOME Anahtarlığı: Gizli Hizmet
+Comment[ug]=گىنوم ئاچقۇچ ھالقىسى：مەخپىيەت مۇلازىمىتى
+Comment[uk]=Служба в'язки ключів GNOME: служба шифрування
+Comment[vi]=Chùm chìa khóa GNOME: dịch vụ bảo mật
+Comment[zh_CN]=GNOME 密钥环：保密服务
+Comment[zh_HK]=GNOME 密碼匙圈：保密服務
+Comment[zh_TW]=GNOME 鑰匙圈：保密服務
+Comment=GNOME Keyring: Secret Service
+Exec=@keyring@/gnome-keyring-daemon --start --components=secrets
+OnlyShowIn=GNOME;Unity;MATE;Pantheon;
+NoDisplay=true
+X-GNOME-Autostart-Phase=PreDisplayServer
+X-GNOME-AutoRestart=false
+X-GNOME-Autostart-Notify=true
+X-GNOME-Bugzilla-Bugzilla=GNOME
+X-GNOME-Bugzilla-Product=gnome-keyring
+X-GNOME-Bugzilla-Component=general
+X-GNOME-Bugzilla-Version=3.28.2
diff --git a/autostart/gnome-keyring-ssh-pantheon.desktop b/autostart/gnome-keyring-ssh-pantheon.desktop
deleted file mode 100644
index 3fb9356..0000000
--- a/autostart/gnome-keyring-ssh-pantheon.desktop
+++ /dev/null
@@ -1,15 +0,0 @@
-[Desktop Entry]
-Type=Application
-Name=SSH Key Agent
-Comment=GNOME Keyring: SSH Agent
-Exec=/usr/bin/gnome-keyring-daemon --start --components=ssh
-OnlyShowIn=Pantheon;
-X-GNOME-Autostart-Phase=Initialization
-X-GNOME-AutoRestart=false
-X-GNOME-Autostart-Notify=true
-X-GNOME-Bugzilla-Bugzilla=GNOME
-X-GNOME-Bugzilla-Product=gnome-keyring
-X-GNOME-Bugzilla-Component=general
-X-GNOME-Bugzilla-Version=3.10.1
-NoDisplay=true
-X-Ubuntu-Gettext-Domain=gnome-keyring
diff --git a/autostart/gnome-keyring-ssh-patheon.desktop b/autostart/gnome-keyring-ssh-patheon.desktop
new file mode 100644
index 0000000..b35856f
--- /dev/null
+++ b/autostart/gnome-keyring-ssh-patheon.desktop
@@ -0,0 +1,152 @@
+[Desktop Entry]
+Type=Application
+Name[af]=SSH-sleutelagent
+Name[ar]=عميل مفاتيح SSH
+Name[as]=SSH কি সহায়ক
+Name[ast]=Axente de claves SSH
+Name[be]=SSH-агент ключоў
+Name[bg]=Агентът на SSH за ключове
+Name[bn]=SSH কী এজেন্ট
+Name[bn_IN]=SSH কি এজেন্ট
+Name[bs]=Agent za SSH ključeve
+Name[ca]=Agent de claus SSH
+Name[ca@valencia]=Agent de claus SSH
+Name[cs]=Agent klíčů SSH
+Name[da]=SSH-nøgleagent
+Name[de]=SSH-Schlüsselagent
+Name[el]=Πράκτορας κλειδιού SSH
+Name[en_GB]=SSH Key Agent
+Name[es]=Agente de claves SSH
+Name[et]=SSH-võtmete agent
+Name[eu]=SSH gako-agentea
+Name[fa]=مامور کلید SSH
+Name[fi]=SSH-avainten agentti
+Name[fr]=Agent de clés SSH
+Name[fur]=Agjent clâfs SSH
+Name[gd]=Àidseant iuchair SSH
+Name[gl]=Axente de chave SSH
+Name[gu]=SSH કી ઍજન્ટ
+Name[he]=סוכן מפתחות SSH
+Name[hi]=AFS कुँजी प्रतिनिधि
+Name[hr]=Agent SSH ključa
+Name[hu]=SSH-kulcs ügynök
+Name[id]=Agen Kunci SSH
+Name[it]=Agente chiavi SSH
+Name[ja]=SSH 鍵エージェント
+Name[kk]=SSH кілттер агенті
+Name[km]=ភ្នាក់ងារ​សោ SSH
+Name[kn]=SSH ಕೀಲಿ ಮಧ್ಯವರ್ತಿ
+Name[ko]=SSH 키 에이전트
+Name[lt]=SSH raktų tarnyba
+Name[lv]=SSH atslēgu aģents
+Name[mk]=Агент за SSH клучеви
+Name[ml]=എസ്എസ്എച് കീ ഏജന്റ്
+Name[mr]=SSH कि एजंट
+Name[ms]=Agen Kekunci SSH
+Name[nb]=SSH-nøkkelagent
+Name[nl]=SSH-sleutelagent
+Name[nn]=Nøkkelagent for SSH
+Name[oc]=Agent de claus SSH
+Name[or]=SSH କି ସଦସ୍ୟ
+Name[pa]=SSH ਕੁੰਜੀ ਏਜੰਟ
+Name[pl]=Agent kluczy SSH
+Name[pt]=Agente de chaves SSH
+Name[pt_BR]=Agente de chaves SSH
+Name[ro]=Agent pentru chei SSH
+Name[ru]=Агент ключей SSH
+Name[sk]=Agent kľúčov SSH
+Name[sl]=Agent ključev SSH
+Name[sr]=Агент за ССХ кључеве
+Name[sr@latin]=Agent za SSH ključeve
+Name[sv]=SSH-nyckelagent
+Name[ta]=SSH முகவர்
+Name[te]=SSH కీ వాహకము
+Name[th]=เอเจนต์กุญแจ SSH
+Name[tr]=SSH Anahtar Aracı
+Name[ug]=SSH  ئاچقۇچ ياردەمچىسى
+Name[uk]=Агент ключів SSH
+Name[vi]=Đại diện khóa SSH
+Name[zh_CN]=SSH 密钥代理
+Name[zh_HK]=SSH 密碼匙代理程式
+Name[zh_TW]=SSH 金鑰代理程式
+Name=SSH Key Agent
+Comment[af]=GNOME-sleutelring: SSH-agent
+Comment[ar]=حلقة مفاتيح جنوم: عميل SSH
+Comment[as]=GNOME Keyring: SSH সহায়ক
+Comment[ast]=Depósitu de claves de GNOME: axente SSH
+Comment[be]=Вязкі ключоў GNOME: SSH-агент
+Comment[bg]=Ключодържател на GNOME: агент на SSH
+Comment[bn]=GNOME কী রিং: SSH এজেন্ট
+Comment[bn_IN]=GNOME Keyring: SSH এজেন্ট
+Comment[bs]=Gnomovi privjesci: SSH agent
+Comment[ca]=Anell de claus del GNOME: agent SSH
+Comment[ca@valencia]=Anell de claus del GNOME: agent SSH
+Comment[cs]=Klíčenka GNOME: Agent SSH
+Comment[da]=GNOME-nøgleringsdæmon: SSH-agent
+Comment[de]=GNOME-Schlüsselbunddienst: SSH-Agent
+Comment[el]=Κλειδοθήκη GNOME: Πράκτορας SSH
+Comment[en_GB]=GNOME Keyring: SSH Agent
+Comment[es]=Depósito de claves de GNOME: agente SSH
+Comment[et]=GNOME võtmerõngas: SSH-agent
+Comment[eu]=GNOMEren gako-sorta: SSH agentea
+Comment[fa]=دسته‌کلید گنوم: مامور SSH
+Comment[fi]=Gnomen avainnippu: SSH-agentti
+Comment[fr]=Trousseau de clés de GNOME : agent SSH
+Comment[fur]=Puarteclâfs di GNOME: agjent SSH
+Comment[gd]=Dul-iuchrach: Àidseant SSH
+Comment[gl]=GNOME Keyring: Axente SSH
+Comment[gu]=GNOME કીરીંગ: SSH ઍજન્ટ
+Comment[he]=קבוצת מפתחות של GNOME: סוכן SSH
+Comment[hi]=गनोम कीरिंग: SSH प्रतिनिधि
+Comment[hr]=GNOME skup ključeva: SSH agent
+Comment[hu]=GNOME kulcstartó – SSH-ügynök
+Comment[id]=Ring Kunci GNOME: Agen SSH
+Comment[it]=Portachiavi di GNOME: agente SSH
+Comment[ja]=GNOME キーリング: SSH エージェント
+Comment[kk]=GNOME Keyring: SSH агенті
+Comment[km]=GNOME Keyring ៖ ភ្នាក់ងារ SSH
+Comment[kn]=GNOME ಕೀಲಿಗೊಂಚಲು: SSH ಮಧ್ಯವರ್ತಿ
+Comment[ko]=그놈 키 모음: SSH 에이전트
+Comment[lt]=GNOME raktinė: SSH tarnyba
+Comment[lv]=GNOME atslēgu saišķis — SSH aģents
+Comment[mk]=Приврзок на GNOME: SSH агент
+Comment[ml]=ഗ്നോം കീറിങ്: എസ്എസ്എച് ഏജന്റ്
+Comment[mr]=GNOME किरिंग: SSH एजंट
+Comment[ms]=GNOME Keyring: Agen SSH
+Comment[nb]=GNOME nøkkelring: SSH-agent
+Comment[ne]=जिनोम किरिङ : SSH एजेन्ट
+Comment[nl]=Sleutelbos-service: SSH-agent
+Comment[nn]=GNOME Nøkkelring: SSH-agent
+Comment[oc]=Trossèl de claus GNOME : agent SSH
+Comment[or]=GNOME କି ରିଙ୍ଗ: SSH ସଦସ୍ୟ
+Comment[pa]=ਗਨੋਮ ਕੀਰਿੰਗ: SSH ਏਜੰਟ
+Comment[pl]=Baza kluczy dla środowiska GNOME: agent SSH
+Comment[pt]=GNOME Keyring: agente SSH
+Comment[pt_BR]=Chaveiro do GNOME: Agente SSH
+Comment[ro]=Inelul de chei GNOME: Agent SSH
+Comment[ru]=Связка ключей GNOME: SSH-агент
+Comment[sk]=SSH agent zväzku kľúčov GNOME
+Comment[sl]=Zbirka ključev GNOME: agent SSH
+Comment[sr]=Гномови привесци: ССХ агент
+Comment[sr@latin]=Gnomovi privesci: SSH agent
+Comment[sv]=GNOME-nyckelring: SSH-agent
+Comment[ta]=GNOME கீரிங்: SSH முகவர்
+Comment[te]=GNOME కీరింగ్: SSH వాహకం
+Comment[th]=พวงกุญแจของ GNOME: เอเจนต์ SSH
+Comment[tr]=GNOME Anahtarlığı: SSH Aracı
+Comment[ug]=گىنوم ئاچقۇچ ھالقىسى：SSH  ياردەمچىسى
+Comment[uk]=Служба в'язки ключів GNOME: агент SSH
+Comment[vi]=Chùm chìa khóa GNOME: Đại diện SSH
+Comment[zh_CN]=GNOME 密钥环：SSH 代理
+Comment[zh_HK]=GNOME 密碼匙圈：SSH 代理程式
+Comment[zh_TW]=GNOME 鑰匙圈：SSH 代理程式
+Comment=GNOME Keyring: SSH Agent
+Exec=@keyring@/gnome-keyring-daemon --start --components=ssh
+OnlyShowIn=GNOME;Unity;MATE;Pantheon;
+X-GNOME-Autostart-Phase=PreDisplayServer
+X-GNOME-AutoRestart=false
+X-GNOME-Autostart-Notify=true
+X-GNOME-Bugzilla-Bugzilla=GNOME
+X-GNOME-Bugzilla-Product=gnome-keyring
+X-GNOME-Bugzilla-Component=general
+X-GNOME-Bugzilla-Version=3.28.2
diff --git a/autostart/onboard-autostart-pantheon.desktop b/autostart/onboard-autostart-pantheon.desktop
deleted file mode 100644
index a797f55..0000000
--- a/autostart/onboard-autostart-pantheon.desktop
+++ /dev/null
@@ -1,9 +0,0 @@
-[Desktop Entry]
-Name=Onboard
-Exec=onboard --not-show-in=GNOME
-Type=Application
-NoDisplay=true
-X-Ubuntu-Gettext-Domain=onboard
-AutostartCondition=GSettings org.gnome.desktop.a11y.applications screen-keyboard-enabled
-X-GNOME-AutoRestart=true
-OnlyShowIn=Pantheon;
diff --git a/autostart/orca-autostart-pantheon.desktop b/autostart/orca-autostart-pantheon.desktop
deleted file mode 100644
index 3579138..0000000
--- a/autostart/orca-autostart-pantheon.desktop
+++ /dev/null
@@ -1,10 +0,0 @@
-[Desktop Entry]
-Type=Application
-Name=Orca screen reader
-Exec=orca
-NoDisplay=true
-AutostartCondition=GSettings org.gnome.desktop.a11y.applications screen-reader-enabled
-X-GNOME-AutoRestart=true
-#X-GNOME-Autostart-Phase=Initialization
-OnlyShowIn=Pantheon;
-X-Ubuntu-Gettext-Domain=orca
diff --git a/autostart/org.gnome.SettingsDaemon.A11ySettings-pantheon.desktop b/autostart/org.gnome.SettingsDaemon.A11ySettings.desktop
similarity index 79%
rename from autostart/org.gnome.SettingsDaemon.A11ySettings-pantheon.desktop
rename to autostart/org.gnome.SettingsDaemon.A11ySettings.desktop
index 33204a3..d76166b 100644
--- a/autostart/org.gnome.SettingsDaemon.A11ySettings-pantheon.desktop
+++ b/autostart/org.gnome.SettingsDaemon.A11ySettings.desktop
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Type=Application
 Name=GNOME Settings Daemon's a11y-settings plugin
-Exec=/usr/lib/gnome-settings-daemon/gsd-a11y-settings
+Exec=@settings@/gsd-a11y-settings
 OnlyShowIn=Pantheon;
 NoDisplay=true
 X-GNOME-Autostart-Phase=Initialization
diff --git a/autostart/org.gnome.SettingsDaemon.Clipboard-pantheon.desktop b/autostart/org.gnome.SettingsDaemon.Clipboard.desktop
similarity index 80%
rename from autostart/org.gnome.SettingsDaemon.Clipboard-pantheon.desktop
rename to autostart/org.gnome.SettingsDaemon.Clipboard.desktop
index 07a7797..7f8e286 100644
--- a/autostart/org.gnome.SettingsDaemon.Clipboard-pantheon.desktop
+++ b/autostart/org.gnome.SettingsDaemon.Clipboard.desktop
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Type=Application
 Name=GNOME Settings Daemon's clipboard plugin
-Exec=/usr/lib/gnome-settings-daemon/gsd-clipboard
+Exec=@settings@/gsd-clipboard
 OnlyShowIn=Pantheon;
 NoDisplay=true
 X-GNOME-Autostart-Phase=Initialization
diff --git a/autostart/org.gnome.SettingsDaemon.Color-pantheon.desktop b/autostart/org.gnome.SettingsDaemon.Color.desktop
similarity index 81%
rename from autostart/org.gnome.SettingsDaemon.Color-pantheon.desktop
rename to autostart/org.gnome.SettingsDaemon.Color.desktop
index b267495..485ed2e 100644
--- a/autostart/org.gnome.SettingsDaemon.Color-pantheon.desktop
+++ b/autostart/org.gnome.SettingsDaemon.Color.desktop
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Type=Application
 Name=GNOME Settings Daemon's color plugin
-Exec=/usr/lib/gnome-settings-daemon/gsd-color
+Exec=@settings@/gsd-color
 OnlyShowIn=Pantheon;
 NoDisplay=true
 X-GNOME-Autostart-Phase=Initialization
diff --git a/autostart/org.gnome.SettingsDaemon.Datetime-pantheon.desktop b/autostart/org.gnome.SettingsDaemon.Datetime.desktop
similarity index 80%
rename from autostart/org.gnome.SettingsDaemon.Datetime-pantheon.desktop
rename to autostart/org.gnome.SettingsDaemon.Datetime.desktop
index 16539b4..1756938 100644
--- a/autostart/org.gnome.SettingsDaemon.Datetime-pantheon.desktop
+++ b/autostart/org.gnome.SettingsDaemon.Datetime.desktop
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Type=Application
 Name=GNOME Settings Daemon's datetime plugin
-Exec=/usr/lib/gnome-settings-daemon/gsd-datetime
+Exec=@settings@/gsd-datetime
 OnlyShowIn=Pantheon;
 NoDisplay=true
 X-GNOME-Autostart-Phase=Initialization
diff --git a/autostart/org.gnome.SettingsDaemon.Housekeeping-pantheon.desktop b/autostart/org.gnome.SettingsDaemon.Housekeeping.desktop
similarity index 80%
rename from autostart/org.gnome.SettingsDaemon.Housekeeping-pantheon.desktop
rename to autostart/org.gnome.SettingsDaemon.Housekeeping.desktop
index 961c07e..4c954c0 100644
--- a/autostart/org.gnome.SettingsDaemon.Housekeeping-pantheon.desktop
+++ b/autostart/org.gnome.SettingsDaemon.Housekeeping.desktop
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Type=Application
 Name=GNOME Settings Daemon's housekeeping plugin
-Exec=/usr/lib/gnome-settings-daemon/gsd-housekeeping
+Exec=@settings@/gsd-housekeeping
 OnlyShowIn=Pantheon;
 NoDisplay=true
 X-GNOME-Autostart-Phase=Initialization
diff --git a/autostart/org.gnome.SettingsDaemon.Keyboard-pantheon.desktop b/autostart/org.gnome.SettingsDaemon.Keyboard.desktop
similarity index 80%
rename from autostart/org.gnome.SettingsDaemon.Keyboard-pantheon.desktop
rename to autostart/org.gnome.SettingsDaemon.Keyboard.desktop
index 04136e9..e2070e0 100644
--- a/autostart/org.gnome.SettingsDaemon.Keyboard-pantheon.desktop
+++ b/autostart/org.gnome.SettingsDaemon.Keyboard.desktop
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Type=Application
 Name=GNOME Settings Daemon's keyboard plugin
-Exec=/usr/lib/gnome-settings-daemon/gsd-keyboard
+Exec=@settings@/gsd-keyboard
 OnlyShowIn=Pantheon;
 NoDisplay=true
 X-GNOME-Autostart-Phase=Initialization
diff --git a/autostart/org.gnome.SettingsDaemon.MediaKeys-pantheon.desktop b/autostart/org.gnome.SettingsDaemon.MediaKeys.desktop
similarity index 80%
rename from autostart/org.gnome.SettingsDaemon.MediaKeys-pantheon.desktop
rename to autostart/org.gnome.SettingsDaemon.MediaKeys.desktop
index 5a0b067..6fb4dfe 100644
--- a/autostart/org.gnome.SettingsDaemon.MediaKeys-pantheon.desktop
+++ b/autostart/org.gnome.SettingsDaemon.MediaKeys.desktop
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Type=Application
 Name=GNOME Settings Daemon's media-keys plugin
-Exec=/usr/lib/gnome-settings-daemon/gsd-media-keys
+Exec=@settings@/gsd-media-keys
 OnlyShowIn=Pantheon;
 NoDisplay=true
 X-GNOME-Autostart-Phase=Initialization
diff --git a/autostart/org.gnome.SettingsDaemon.Mouse-pantheon.desktop b/autostart/org.gnome.SettingsDaemon.Mouse.desktop
similarity index 81%
rename from autostart/org.gnome.SettingsDaemon.Mouse-pantheon.desktop
rename to autostart/org.gnome.SettingsDaemon.Mouse.desktop
index f459ec4..18dd3db 100644
--- a/autostart/org.gnome.SettingsDaemon.Mouse-pantheon.desktop
+++ b/autostart/org.gnome.SettingsDaemon.Mouse.desktop
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Type=Application
 Name=GNOME Settings Daemon's mouse plugin
-Exec=/usr/lib/gnome-settings-daemon/gsd-mouse
+Exec=@settings@/gsd-mouse
 OnlyShowIn=Pantheon;
 NoDisplay=true
 X-GNOME-Autostart-Phase=Initialization
diff --git a/autostart/org.gnome.SettingsDaemon.Power-pantheon.desktop b/autostart/org.gnome.SettingsDaemon.Power.desktop
similarity index 81%
rename from autostart/org.gnome.SettingsDaemon.Power-pantheon.desktop
rename to autostart/org.gnome.SettingsDaemon.Power.desktop
index e7d5a9c..ab809d9 100644
--- a/autostart/org.gnome.SettingsDaemon.Power-pantheon.desktop
+++ b/autostart/org.gnome.SettingsDaemon.Power.desktop
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Type=Application
 Name=GNOME Settings Daemon's power plugin
-Exec=/usr/lib/gnome-settings-daemon/gsd-power
+Exec=@settings@/gsd-power
 OnlyShowIn=Pantheon;
 NoDisplay=true
 X-GNOME-Autostart-Phase=Initialization
diff --git a/autostart/org.gnome.SettingsDaemon.PrintNotifications-pantheon.desktop b/autostart/org.gnome.SettingsDaemon.PrintNotifications.desktop
similarity index 78%
rename from autostart/org.gnome.SettingsDaemon.PrintNotifications-pantheon.desktop
rename to autostart/org.gnome.SettingsDaemon.PrintNotifications.desktop
index 57d3d05..c048a13 100644
--- a/autostart/org.gnome.SettingsDaemon.PrintNotifications-pantheon.desktop
+++ b/autostart/org.gnome.SettingsDaemon.PrintNotifications.desktop
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Type=Application
 Name=GNOME Settings Daemon's print-notifications plugin
-Exec=/usr/lib/gnome-settings-daemon/gsd-print-notifications
+Exec=@settings@/gsd-print-notifications
 OnlyShowIn=Pantheon;
 NoDisplay=true
 X-GNOME-Autostart-Phase=Initialization
diff --git a/autostart/org.gnome.SettingsDaemon.Rfkill-pantheon.desktop b/autostart/org.gnome.SettingsDaemon.Rfkill.desktop
similarity index 81%
rename from autostart/org.gnome.SettingsDaemon.Rfkill-pantheon.desktop
rename to autostart/org.gnome.SettingsDaemon.Rfkill.desktop
index dbdcf3b..1e6a51f 100644
--- a/autostart/org.gnome.SettingsDaemon.Rfkill-pantheon.desktop
+++ b/autostart/org.gnome.SettingsDaemon.Rfkill.desktop
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Type=Application
 Name=GNOME Settings Daemon's rfkill plugin
-Exec=/usr/lib/gnome-settings-daemon/gsd-rfkill
+Exec=@settings@/gsd-rfkill
 OnlyShowIn=Pantheon;
 NoDisplay=true
 X-GNOME-Autostart-Phase=Initialization
diff --git a/autostart/org.gnome.SettingsDaemon.Sharing-pantheon.desktop b/autostart/org.gnome.SettingsDaemon.Sharing.desktop
similarity index 81%
rename from autostart/org.gnome.SettingsDaemon.Sharing-pantheon.desktop
rename to autostart/org.gnome.SettingsDaemon.Sharing.desktop
index 5a21ddf..ae7748c 100644
--- a/autostart/org.gnome.SettingsDaemon.Sharing-pantheon.desktop
+++ b/autostart/org.gnome.SettingsDaemon.Sharing.desktop
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Type=Application
 Name=GNOME Settings Daemon's sharing plugin
-Exec=/usr/lib/gnome-settings-daemon/gsd-sharing
+Exec=@settings@/gsd-sharing
 OnlyShowIn=Pantheon;
 NoDisplay=true
 X-GNOME-Autostart-Phase=Initialization
diff --git a/autostart/org.gnome.SettingsDaemon.Smartcard-pantheon.desktop b/autostart/org.gnome.SettingsDaemon.Smartcard.desktop
similarity index 80%
rename from autostart/org.gnome.SettingsDaemon.Smartcard-pantheon.desktop
rename to autostart/org.gnome.SettingsDaemon.Smartcard.desktop
index 37197b1..8d906d3 100644
--- a/autostart/org.gnome.SettingsDaemon.Smartcard-pantheon.desktop
+++ b/autostart/org.gnome.SettingsDaemon.Smartcard.desktop
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Type=Application
 Name=GNOME Settings Daemon's smartcard plugin
-Exec=/usr/lib/gnome-settings-daemon/gsd-smartcard
+Exec=@settings@/gsd-smartcard
 OnlyShowIn=Pantheon;
 NoDisplay=true
 X-GNOME-Autostart-Phase=Initialization
diff --git a/autostart/org.gnome.SettingsDaemon.Sound-pantheon.desktop b/autostart/org.gnome.SettingsDaemon.Sound.desktop
similarity index 81%
rename from autostart/org.gnome.SettingsDaemon.Sound-pantheon.desktop
rename to autostart/org.gnome.SettingsDaemon.Sound.desktop
index b8c4cf8..214d9b8 100644
--- a/autostart/org.gnome.SettingsDaemon.Sound-pantheon.desktop
+++ b/autostart/org.gnome.SettingsDaemon.Sound.desktop
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Type=Application
 Name=GNOME Settings Daemon's sound plugin
-Exec=/usr/lib/gnome-settings-daemon/gsd-sound
+Exec=@settings@/gsd-sound
 OnlyShowIn=Pantheon;
 NoDisplay=true
 X-GNOME-Autostart-Phase=Initialization
diff --git a/autostart/org.gnome.SettingsDaemon.Wacom-pantheon.desktop b/autostart/org.gnome.SettingsDaemon.Wacom.desktop
similarity index 81%
rename from autostart/org.gnome.SettingsDaemon.Wacom-pantheon.desktop
rename to autostart/org.gnome.SettingsDaemon.Wacom.desktop
index c9d8496..49d0af9 100644
--- a/autostart/org.gnome.SettingsDaemon.Wacom-pantheon.desktop
+++ b/autostart/org.gnome.SettingsDaemon.Wacom.desktop
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Type=Application
 Name=GNOME Settings Daemon's wacom plugin
-Exec=/usr/lib/gnome-settings-daemon/gsd-wacom
+Exec=@settings@/gsd-wacom
 OnlyShowIn=Pantheon;
 NoDisplay=true
 X-GNOME-Autostart-Phase=Initialization
diff --git a/autostart/org.gnome.SettingsDaemon.XSettings-pantheon.desktop b/autostart/org.gnome.SettingsDaemon.XSettings.desktop
similarity index 80%
rename from autostart/org.gnome.SettingsDaemon.XSettings-pantheon.desktop
rename to autostart/org.gnome.SettingsDaemon.XSettings.desktop
index 78acfce..5ee6cc2 100644
--- a/autostart/org.gnome.SettingsDaemon.XSettings-pantheon.desktop
+++ b/autostart/org.gnome.SettingsDaemon.XSettings.desktop
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Type=Application
 Name=GNOME Settings Daemon's xsettings plugin
-Exec=/usr/lib/gnome-settings-daemon/gsd-xsettings
+Exec=@settings@/gsd-xsettings
 OnlyShowIn=Pantheon;
 NoDisplay=true
 X-GNOME-Autostart-Phase=Initialization
diff --git a/autostart/user-dirs-update-gtk-pantheon.desktop b/autostart/user-dirs-update-gtk-pantheon.desktop
deleted file mode 100644
index eba3326..0000000
--- a/autostart/user-dirs-update-gtk-pantheon.desktop
+++ /dev/null
@@ -1,12 +0,0 @@
-[Desktop Entry]
-Encoding=UTF-8
-Exec=xdg-user-dirs-gtk-update
-Name=User folders update
-Comment=Update common folders names to match current locale
-Terminal=false
-OnlyShowIn=Pantheon;
-Type=Application
-StartupNotify=false
-X-KDE-autostart-after=panel
-NoDisplay=true
-X-Ubuntu-Gettext-Domain=xdg-user-dirs-gtk
