diff --git a/CMakeLists.txt b/CMakeLists.txt
index 37faa88..3df6e1a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,7 +11,7 @@ set (GSETTINGSDIR "${DATADIR}/glib-2.0/schemas/")
 set (PKGDATADIR "${DATADIR}/${CMAKE_PROJECT_NAME}")
 set (GETTEXT_PACKAGE "${CMAKE_PROJECT_NAME}")
 set (RELEASE_NAME "Preferences with piazzaz.")
-set (PLUGS_DIR ${CMAKE_INSTALL_FULL_LIBDIR}/${CMAKE_PROJECT_NAME})
+set (PLUGS_DIR "/run/current-system/sw/lib/switchboard")
 set (VERSION "2.3.1")
 set (VERSION_INFO "Stable Release")
 
@@ -147,8 +147,8 @@ add_subdirectory (schemas)
 install (FILES ${CMAKE_CURRENT_SOURCE_DIR}/schemas/io.elementary.switchboard.gschema.xml DESTINATION ${GSETTINGSDIR})
 
 # Create the standard directories to insert plugs.
-install(DIRECTORY DESTINATION ${PLUGS_DIR})
-install(DIRECTORY DESTINATION ${PLUGS_DIR}/hardware)
-install(DIRECTORY DESTINATION ${PLUGS_DIR}/network)
-install(DIRECTORY DESTINATION ${PLUGS_DIR}/personal)
-install(DIRECTORY DESTINATION ${PLUGS_DIR}/system)
+#install(DIRECTORY DESTINATION ${PLUGS_DIR})
+#install(DIRECTORY DESTINATION ${PLUGS_DIR}/hardware)
+#install(DIRECTORY DESTINATION ${PLUGS_DIR}/network)
+#install(DIRECTORY DESTINATION ${PLUGS_DIR}/personal)
+#install(DIRECTORY DESTINATION ${PLUGS_DIR}/system)
diff --git a/lib/switchboard.pc.cmake b/lib/switchboard.pc.cmake
index 613e92e..917c492 100644
--- a/lib/switchboard.pc.cmake
+++ b/lib/switchboard.pc.cmake
@@ -1,8 +1,8 @@
 prefix=@PREFIX@
 exec_prefix=${prefix}
-libdir=${prefix}/@CMAKE_INSTALL_LIBDIR@
-includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@
-plugsdir=${libdir}/switchboard
+libdir=@CMAKE_INSTALL_LIBDIR@
+includedir=@CMAKE_INSTALL_INCLUDEDIR@
+plugsdir=/run/current-system/sw/lib/switchboard
  
 Name: Switchboard
 Description: Switchboard headers
