--- a/build-aux/meson/post_install.py
+++ b/build-aux/meson/post_install.py
@@ -17,9 +17,9 @@ if 'DESTDIR' not in os.environ:
     subprocess.call(['gtk-update-icon-cache', '-qtf',
                      os.path.join(datadir, 'icons', 'hicolor')])
 
-    print('Updating desktop database...')
-    subprocess.call(['update-desktop-database', '-q',
-                     os.path.join(datadir, 'applications')])
+    # print('Updating desktop database...')
+    # subprocess.call(['update-desktop-database', '-q',
+    #                  os.path.join(datadir, 'applications')])
 
     print('Compiling python modules...')
     subprocess.call([sys.executable, '-m', 'compileall', '-f', '-q',
